samples/interactive-tutorials/src/main/java/events/ImportUserEventsBigQuery.java

```diff
@@ -0,0 +1,107 @@
+    String projectId = System.getenv("PROJECT_ID");
+    String defaultCatalog =
+        String.format("projects/%s/locations/global/catalogs/default_catalog", projectId);
+    // TO CHECK ERROR HANDLING PASTE THE INVALID CATALOG NAME HERE: defaultCatalog =
+    // "invalid_catalog_name"
+
+    importUserEventsFromBigQuery(projectId, defaultCatalog);
+  }
+
+  public static void importUserEventsFromBigQuery(String projectId, String defaultCatalog)
+      throws IOException, InterruptedException {
+    String datasetId = "user_events";
+    String tableId = "events";
+    // TO CHECK ERROR HANDLING USE THE TABLE OF INVALID USER EVENTS: tableId = "events_some_invalid"
+    String dataSchema = "user_event";
+
+    BigQuerySource bigQuerySource =
+        BigQuerySource.newBuilder()
+            .setProjectId(projectId)
+            .setDatasetId(datasetId)
+            .setTableId(tableId)
+            .setDataSchema(dataSchema)
+            .build();
+
+    UserEventInputConfig inputConfig =
+        UserEventInputConfig.newBuilder().setBigQuerySource(bigQuerySource).build();
+
+    ImportUserEventsRequest importRequest =
+        ImportUserEventsRequest.newBuilder()
+            .setParent(defaultCatalog)
+            .setInputConfig(inputConfig)
+            .build();
+
+    System.out.printf("Import user events from BigQuery source request: %s%n", importRequest);
+
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the "close" method on the client to safely clean up any remaining background resources.
+    try (UserEventServiceClient serviceClient = UserEventServiceClient.create()) {
+      String operationName = serviceClient.importUserEventsCallable().call(importRequest).getName();
+
+      System.out.printf("OperationName = %s\n", operationName);
+      OperationsClient operationsClient = serviceClient.getOperationsClient();
+      Operation operation = operationsClient.getOperation(operationName);
+
+      while (!operation.getDone()) {
+        // Keep polling the operation periodically until the import task is done.
+        int awaitDuration = 30000;
+        Thread.sleep(awaitDuration);
+        operation = operationsClient.getOperation(operationName);
+      }
+
+      if (operation.hasMetadata()) {
+        ImportMetadata metadata = operation.getMetadata().unpack(ImportMetadata.class);
+        System.out.printf(
+            "Number of successfully imported events: %s\n", metadata.getSuccessCount());
+        System.out.printf(
+            "Number of failures during the importing: %s\n", metadata.getFailureCount());
+      }
+
+      if (operation.hasResponse()) {
+        ImportUserEventsResponse response =
+            operation.getResponse().unpack(ImportUserEventsResponse.class);
+        System.out.printf("Operation result: %s%n", response);
+      }
+    }
+  }
```

Comment:
I don't see any error/exception handling logic... What kind of exception should be caught? An [example](https://github.com/googleapis/java-bigquery/blob/001302bbfd3d1e071bb48ad2a783183fd48f6059/samples/snippets/src/main/java/com/example/bigquery/CreateDataset.java#L45-L47).

Reply:
Fixed.